src/main/scala/viper/gobra/frontend/info/implementation/property/Assignability.scala

```diff
@@ -159,6 +166,41 @@ trait Assignability extends BaseProperty { this: TypeInfoImpl =>
     case (l, t) => failedProp(s"cannot assign literal $l to $t")
   }
 
+  def assignWithinBounds: Property[(Type, PExpression)] = createFlatProperty[(Type, PExpression)] {
+    case (typ, expr) => s"constant expression $expr overflows $typ"
+  } {
+    case (typ, expr) =>
+      val constVal = intConstantEval(expr)
+      constVal.isEmpty || intValInBounds(constVal.get, typ)
+  }
+
+  private def intValInBounds(value: BigInt, typ: Type): Boolean =
+    underlyingType(typ) match {
+      case IntT(t) => t match {
+        case UntypedConst => true
+        case x =>
+          val bounds = getBoundsOfIntKind(x)
+          bounds.lower <= value && value <= bounds.upper
+      }
+
+      case _ => ???
+    }
+
+  private val getBoundsOfIntKind: IntegerKind => Bounds = {
```

Comment:
The bounds seem a bit redundant since every kind X is mapped to Bounds.X.

Reply:
You could add the bounds to the type kind already.

Reply:
Removed Bounds.scala file altogether in bceb60f